hail/src/main/scala/is/hail/rvd/RVDPartitionInfo.scala

```diff
@@ -64,15 +66,20 @@ object RVDPartitionInfo {
       while (it.hasNext) {
         val f = it.next()
 
-        if (typ.kOrd.lt(f, prevF.value)) {
+        if (sortedness > UNSORTED && typ.kOrd.lt(f, prevF.value)) {
           if (pkOrd.lt(f, prevF.value)) {
-            if (sortedness > UNSORTED)
-              log.info(s"unsorted: ${ f.pretty(typ.kType) }, ${ prevF.pretty }")
+            val curr = f.pretty(typ.kType)
+            val prev = prevF.pretty
+            log.info(s"unsorted: $curr, $prev")
+            contextStr = s"CURRENT=$curr, PREV=$prev"
             sortedness = UNSORTED
-          } else
-            if (sortedness > TSORTED)
-              log.info(s"tsorted: ${ f.pretty(typ.kType) }, ${ prevF.pretty }")
+          } else if (sortedness > TSORTED) {
+            val curr = f.pretty(typ.kType)
+            val prev = prevF.pretty
+            log.info(s"partition-key-sorted: $curr, $prev")
+            contextStr = s"CURRENT=$curr, PREV=$prev"
```

Comment:
Doesn't this end up logging the out-of-order key pair twice?

Reply:
This logging statement will be in the executor logs for each partition. The logging in the RVD.scala file will be in the Hail log once, for an arbitrarily chosen min-sortedness partition.